### PR TITLE
RFC6587 et. al

### PR DESCRIPTION
**Pull request by leodido**
This PR introduces several features and aspects:

- [x] Go modules
- [x] Interfaces
    - [x] **syslog.Machine**s => parsers a byte slice
    - [x] **syslog.Parser**s => parsers supporting a stream
- [x] Streaming impl. of non-transparent framing for syslog transports, ie., [RFC6587](https://tools.ietf.org/html/rfc6587) (refs #16)
    - [x] Documentation
    - [ ] Error handling
- [x] Refactoring of existing RFC5424 to match interfaces and conventions
- [x] Refactoring of existing RFC5425 to match interfaces and conventions
- [x] CI setup

Note that I'll introduce **BC** with this PR.

### CODE REVIEW

File: rfc5424/machine_test.go [hunk 1]
```diff
@@ -52,7 +53,7 @@ var testCases = []testCase{
 	// Invalid, multiple syslog messages on multiple lines
 	{
 		[]byte(`<1>1 - - - - - -
-		<2>1 - - - - - -`),
+			<2>1 - - - - - -`),
```
Comment (leodido) [inline rfc5424/machine_test.go]: Memo: remove tab here.

File: rfc6587/parser.go.rl [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+// Exec implements the ragel.Parser interface.
+func (m *machine) Exec(s *parser.State) (int, int) {
+    // Retrieve previously stored parsing variables
+    cs, p, pe, eof, data := s.Get()
+    // Inline FSM code here
+    %% write exec;
+    // Update parsing variables
+	s.Set(cs, p, pe, eof)
```
Comment (goller) [inline rfc6587/parser.go.rl]: look like 2 vs 4 spaces here

File: rfc6587/parser.go.rl [hunk 3]
```diff
@@ -0,0 +1,155 @@
+package rfc6587
+
+import (
+    "io"
+    parser "github.com/leodido/ragel-machinery/parser"
+	syslog "github.com/influxdata/go-syslog"
```
Comment (goller) [inline rfc6587/parser.go.rl]: 4 vs 2 spaces here

**Review by goller — APPROVED** [review]
Looks great! It's quite clear how you did the LF\NUL.

Once you would run a formatter over the ragel file, I think it is ready to go.

**Review by danielnelson — APPROVED** [review]

**Comment by danielnelson** [comment]
Make sure to add a license to ragel-machinery.

**Comment by leodido** [comment]
Yup, done. Thanks @danielnelson

**Pull request merged**